### PR TITLE
Add support for image digests in helm charts

### PR DESCRIPTION
**Pull request by aanm**
Read per commit

### CODE REVIEW

File: install/kubernetes/cilium/templates/cilium-agent-daemonset.yaml [hunk 1]
```diff
@@ -193,7 +193,11 @@ spec:
 {{- with .Values.extraEnv }}
 {{ toYaml . | trim | indent 8 }}
 {{- end }}
+{{- if .Values.image.useDigest }}
```
Comment (sayboras) [inline install/kubernetes/cilium/templates/cilium-agent-daemonset.yaml]: any reason not to use .Values.image.digest here directly (i.e. check if .Values.image.digest is not empty) ?

Comment (aanm) [reply install/kubernetes/cilium/templates/cilium-agent-daemonset.yaml]: @sayboras see https://github.com/cilium/cilium/pull/15185#discussion_r586700040 if it makes sense

File: .github/workflows/images-legacy-releases.yaml [hunk 2]
```diff
@@ -16,61 +16,80 @@ jobs:
...
+        id: tag
+        run: |
+          echo ::set-output name=tag::${GITHUB_REF##*/}
+
+      - name: Checkout Source Code
+        uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # v2.3.4
+
       - name: Release Build ${{ matrix.name }}
-        uses: docker/build-push-action@4a531fa5a603bab87dfa56578bd82b28508c9547 # v2.2.2
+        uses: docker/build-push-action@9379083e426e2e84abb80c8c091f5cdeb7d3fd7a # v2.2.2
```
Comment (joestringer) [inline .github/workflows/images-legacy-releases.yaml]: Is this still `v2.2.2`? The sha changed.

**Review by joestringer — APPROVED** [review]

File: install/kubernetes/Makefile [hunk 3]
```diff
@@ -56,14 +56,35 @@ LOGO_PATH := Documentation/images/logo-solo.svg
...
+# FIXME change the useDigest to 'true' once images were released by GH actions
 $(QUICK_HUBBLE_INSTALL) quick-hubble-install: $(shell find cilium/ -type f) update-versions
-	$(QUIET)helm template cilium cilium --namespace=kube-system $(QUICK_HUBBLE_OPTIONS) > $(QUICK_HUBBLE_INSTALL)
-
+	$(QUIET)helm template cilium cilium --namespace=kube-system \
+     --set image.useDigest=false \
+     --set operator.image.useDigest=false \
+     --set preflight.image.useDigest=false \
+     --set hubble.relay.image.useDigest=false \
+     --set clustermesh.apiserver.image.useDigest=false \
```
Comment (joestringer) [inline install/kubernetes/Makefile:76]: Presumably these new parameters should be no-ops because the default is false.

Comment (aanm) [reply install/kubernetes/Makefile:76]: Yes, once we release 1.9, we need to change them to true. (the same does not apply to 1.7 as we already release images based on GH actions)

File: .github/workflows/images-legacy-releases.yaml (hunk 2 above)
Comment (aanm) [reply .github/workflows/images-legacy-releases.yaml]: Not it is not, I need to remove all comments since dependabot will update those SHAs automatically.

**Comment by aanm** [comment]
test-backport-1.9

**Comment by aanm** [comment]
test-1.12-netnext

**Comment by aanm** [comment]
hit https://github.com/cilium/cilium/issues/15244

**Pull request merged**